src/dmtcpplugin.cpp

```diff
@@ -352,7 +352,7 @@ dmtcp_get_restart_env(const char *name,   // IN
     size = statbuf.st_size + 1;
   }
 
-  char env_buf[size] = {0};
+  char *env_buf = static_cast<char *>(malloc(size));
```

Comment:
Just checking on your comment.
 1.  I don't want to use alloca since it's not POSIX.
 2. Is the issue that 'malloc' change the user's malloc arena, possibly doing things like forcing a mmap or interfering with the user's tools to instrument malloc?  (However, JALLOC_MALLOC does do mmap.)

Probably the best thing is just to use JALLOC_MALLOC, so that we can interpose and change it later if we get into trouble.

Does that sound right?

Reply:
@karya0, Thanks for pushing me to think about this more deeply.  I think I found a good compromise.  I declare
``` C
char env_buf_small[10000];
if (size <= sizeiof env_buf_small) {
  env_buf = (char *)env_buf_small;
} else {
  env_buf = JALLOC_MALLOC(size);;
}
```

This seems to be the best of both worlds: directly allocate on stack (no `alloca`) for small allocations, and otherwise use `JALLOC_MALLOC()`.

If you agree with this fix, then I'll push it in.

Reply:
@karya0 , Ping.  I've now changed this to address your concerns.